test/uk/gov/hmrc/transitmovementsrouter/services/XmlParserSpec.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.services
+
+import akka.stream.alpakka.xml.scaladsl._
+import akka.stream.scaladsl.Sink
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures.whenReady
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+import uk.gov.hmrc.transitmovementsrouter.base.StreamTestHelpers._
+import uk.gov.hmrc.transitmovementsrouter.base.TestActorSystem
+import uk.gov.hmrc.transitmovementsrouter.models._
+
+import scala.xml.Utility.trim
+import scala.xml._
+
+class XmlParserSpec extends AnyFreeSpec with TestActorSystem with Matchers {
+
+  "MessageSender parser" - new Setup {
+    "when provided with a valid message" in {
+      val stream       = createParsingEventStream(cc015cValidGB)
+      val parsedResult = stream.via(XmlParser.messageSenderExtractor).runWith(Sink.head)
+
+      whenReady(parsedResult) {
+        _.right.get mustBe MessageSenderId("GB1234")
+      }
+    }
+
+    "when provided with a missing messageSender node" in {
+      val stream       = createParsingEventStream(cc015cNoMessageSenderNode)
+      val parsedResult = stream.via(XmlParser.messageSenderExtractor).runWith(Sink.head)
+
+      whenReady(parsedResult) {
+        _.isLeft
```

Comment:
We should check that the correct error is returned, not just that one is returned.

Reply:
Agree. Will change.